docs/dev/building.md

```diff
@@ -77,20 +84,36 @@ If you don't specify a build mode, the `build.ninja` will contain configuration
 
 [Ninja]: https://ninja.org/
 
+In the CMake-based building system, the build modes are mapped to the configuration names:
```

Comment:
In what sense do we still have the notion of "build modes" taking dev/release and so on? The directory is no longer build/dev, it is build/Dev, and so on. Why shold a user care about the old names?

Reply:
our toolings are built around "dev", "release", etc. and quite a few developers are used to "dev" and "release" notations, so i think i have to note this down.